src/cpp/src/perf_metrics.cpp

```diff
@@ -0,0 +1,118 @@
+            raw_metrics.m_durations[i] /= batch_sizes[i];
+            num_generated_tokens += batch_sizes[i];
+            start_time_val = tok_times[i];
+        }
+    }
+    
+    // calc_mean_and_std will convert microsecond to milliseconds.
+    tpot = calc_mean_and_std(raw_metrics.m_durations);
+    ttft = calc_mean_and_std(raw_metrics.m_times_to_first_token);
+
+    generate_duration = calc_mean_and_std(raw_metrics.generate_durations);
+    generate_duration = calc_mean_and_std(raw_metrics.generate_durations);
+
+    tokenization_duration = calc_mean_and_std(raw_metrics.tokenization_durations);
+    detokenization_duration = calc_mean_and_std(raw_metrics.detokenization_durations);    
+    
+    // tokens per second
+    throughput = {1000.0f / tpot.mean, (tpot.std * 1000.0f) / (tpot.mean * tpot.mean)};
+}
+
+PerfMetrics PerfMetrics::operator+(const PerfMetrics& right) const {
+    OPENVINO_ASSERT(right.load_time == load_time, "generation metrics can be accumulated only for the same pipeline");
+    
+    // Copy left value to res.
+    PerfMetrics res = *this;
+
+    // Concatenate durations, batch_sizes first token times.
+    auto& new_durations = res.raw_metrics.m_durations;
+    auto& new_batch_sizes = res.raw_metrics.m_batch_sizes;
+    auto& new_times_to_first_token = res.raw_metrics.m_times_to_first_token;
+    auto& right_durations = right.raw_metrics.m_durations;
+    auto& right_batch_sizes = right.raw_metrics.m_batch_sizes;
+    auto& right_times_to_first_token = right.raw_metrics.m_times_to_first_token;
+    
+    new_durations.insert(new_durations.end(), right_durations.begin(), right_durations.end());
+    new_times_to_first_token.insert(new_times_to_first_token.end(), right_times_to_first_token.begin(), right_times_to_first_token.end());
+    new_batch_sizes.insert(new_batch_sizes.end(), right_batch_sizes.begin(), right_batch_sizes.end());
+
+    // Concatenate tokenization/detokenization and total generation times.
+    auto& new_tok_durations = res.raw_metrics.tokenization_durations;
+    auto& new_detok_durations = res.raw_metrics.detokenization_durations;
+    auto& new_gen_durations = res.raw_metrics.generate_durations;
+    auto& right_tok_durations = right.raw_metrics.tokenization_durations;
+    auto& right_detok_durations = right.raw_metrics.detokenization_durations;
+    auto& right_gen_durations = right.raw_metrics.generate_durations;
+    
+    new_tok_durations.insert(new_tok_durations.end(), right_tok_durations.begin(), right_tok_durations.end());
+    new_detok_durations.insert(new_detok_durations.end(), right_detok_durations.begin(), right_detok_durations.end());
+    new_gen_durations.insert(new_gen_durations.end(), right_gen_durations.begin(), right_gen_durations.end());
+
+    res.num_generated_tokens = num_generated_tokens + right.num_generated_tokens;
+    res.num_input_tokens = num_generated_tokens + right.num_input_tokens;
+    res.load_time = load_time;
+    res.evaluate_statistics();
```

Comment:
Added getters. Now in order to get user should call `perf_metrics.get_tokenization_duration().mean` if statistics are fresh and already evaluated then it will return values, if they are not fresh getter will call `evaluate_statistics()`